lib/OAuth2.php

```diff
@@ -823,12 +823,8 @@ public function grantAccessToken(Request $request = null)
                 $stored = $this->grantAccessTokenRefreshToken($client, $input);
                 break;
             default:
-                if (filter_var($input["grant_type"], FILTER_VALIDATE_URL)) {
```

Comment:
Maybe it would be better to change this check to keep the uri verification:
```php
if (filter_var($input["grant_type"], FILTER_VALIDATE_URL) || 0 === strpos($input["grant_type"], 'urn')) {
```